### PR TITLE
Enforce datatypes not containing components

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Checking now during codegen that:
* archetypes only contain components
* components only contain primitive data or datatypes
   * this is a relaxation of what I had in mind at first: There's a few components that are currently *not* referring to any rerun datatypes at all but only contain primitive data
* datatypes only contain primitive data or datatypes

In order to pass this, introduces new datatypes which get nested into various components
* Color
* Label
* ClassId
* KeypointId

This is smooth for the most part but causes some friction around class/keypoint id.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2952) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2952)
- [Docs preview](https://rerun.io/preview/pr%3Aandreas%2Fenforce-datatype-separation/docs)
- [Examples preview](https://rerun.io/preview/pr%3Aandreas%2Fenforce-datatype-separation/examples)

### CODE REVIEW

File: crates/re_query/tests/query_tests.rs [hunk 1]
```diff
@@ -175,7 +175,7 @@ fn no_instance_join_query() {
     // │ u64      ┆ struct[2] ┆ u32        │
     // ╞══════════╪═══════════╪════════════╡
     // │ 0        ┆ {1.0,2.0} ┆ 4278190080 │
-    // ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
+    // ├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
```
Comment (jleibs) [inline crates/re_query/tests/query_tests.rs:178]: ?

Comment (Wumpf) [reply crates/re_query/tests/query_tests.rs:178]: There was one too many. Looks nicer now ;)

Comment (Wumpf) [reply crates/re_query/tests/query_tests.rs:178]: Hm okay that's only my editor that doensn't display ┼ correctly -.-

File: crates/re_types/definitions/rerun/components/keypoint_id.fbs [hunk 2]
```diff
@@ -19,13 +19,14 @@ namespace rerun.components;
 /// \rs
 /// \rs Used to look up an `crate::components::AnnotationInfo` for a Keypoint within the `crate::components::AnnotationContext`.
 struct KeypointId (
-  "attr.python.aliases": "float",
```
Comment (jleibs) [inline crates/re_types/definitions/rerun/components/keypoint_id.fbs:22]: :grimacing:

Comment (Wumpf) [reply crates/re_types/definitions/rerun/components/keypoint_id.fbs:22]: yeah no idea where that came from. Suprised that this didn't cause some mypy issues?

**Review by Wumpf — COMMENTED** [review]
lgtm! good teamwork!

**Review by jleibs — APPROVED** [review]

**Pull request merged**